### DISCUSSION TITLE
更新日志 - 22.03.6 - 20240315

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#1420] [Announcements]
**应用过滤更新**

# 系统
1. 基于端口的磁盘自动挂载：修复了VirtIO，SAS或SCSI设备挂载点总是“/mnt/-1”的问题，主要受影响的是虚拟机内的 iStoreOS

# 应用
* 应用过滤（OpenAppFilter/OAF）：
  1. 修复某些网卡过滤效果不好的问题（主要技术原因是OAF官方版本没处理某些网卡产生的非线性SKB）
  2. 支持 IPv6 包过滤，同样也是OAF官方版本目前不支持的功能。（注意旁路模式下可能没什么效果）
  3. 勾选过滤APP以后对已建立的连接进行阻断 （只能处理OAF启动以后建立的连接，因为大部分特征都是在连接建立的头几个包出现）
  4. 更新特征库到官方最新版本，主要增加了微信等特征
  * 注意，防火墙的“软件流量分载”会影响 OAF 过滤效果

# 设备
* rk35xx：内核支持了 TCP BBR 拥塞控制算法，默认不开启，需要测试的话自己用命令打开 `sysctl -w net.ipv4.tcp_congestion_control=bbr` （重启会失效）。注意 BBR 不是灵丹妙药，可能会产生其他问题，例如影响samba速度。（其他非rk35xx机型可以在软件包里面搜索 `kmod-tcp-bbr` 安装就可以使用BBR了，无需命令） #1256

**Comment by xiaotiantian1132** [comment]
R2S当前版本[22.03 branch (git-24.016.46072-3e9fab7](https://github.com/openwrt/luci)，网页升级功能点击可以下载，点升级后执行升级操作，自动重启后版本没变，试了很多次，日志有地方看吗

**Reply by jjm2473** [reply]
版本能发完整的吗？这是luci版本

**Reply by xiaotiantian1132** [reply]
[Powered by LuCI istoreos-22.03 branch (git-24.016.46072-3e9fab7)](https://github.com/openwrt/luci) / [ArgonTheme v2.2.10.10](https://github.com/jerrykuku/luci-theme-argon) / iStoreOS 22.03.6 2024022310

**Reply by jjm2473** [reply]
/overlay/upgrae.log 有升级过程的日志

**Reply by xiaotiantian1132** [reply]
![image](https://github.com/istoreos/istoreos/assets/39047774/68714e9d-ca82-4406-b723-24322355abcf)
全部日志都在这里了，没啥特殊的啊

**Reply by xiaotiantian1132** [reply]
升不了是只能重刷了？

**Reply by a404340821** [reply]
你没有乱升级系统组件吧 根据作者给我的回复我做了测试，执行修复系统组件 可以解决大部分问题

**Reply by xiaotiantian1132** [reply]
系统组件从来没动过，一直是在线更新的，突然这个版本不能往上更新了

**Reply by xiaotiantian1132** [reply]
执行修复系统组件也试过了，还是带branch

**Reply by xiaotiantian1132** [reply]
大佬，给看下吧，想回到能更新的状态

**Reply by jjm2473** [reply]
日志就这些？可能你的系统在升级过程中崩溃了。试试升级之前先关掉clash和docker。如果有ttl调试线，连上去看看。

**Comment by a404340821** [comment]


**Reply by xiaotiantian1132** [reply]
重刷的话，配置还在不？

**Comment by a404340821** [comment]


**Reply by xiaotiantian1132** [reply]
那配置导出，重刷之后导入，能完整保留配置不？

**Comment by a404340821** [comment]


**Reply by jjm2473** [reply]
那说明你自己乱升级软件包了啊，不是早就说过不要乱升级系统软件包吗，而且最近的几个版本已经禁止在页面上升级关键的系统软件包了。恢复出厂设置或者执行修复系统组件， https://github.com/istoreos/istoreos/wiki/FAQ 第2点

还有，你到底回复的是谁，不是邮件发给你就是给你沟通，只是因为你订阅了这个帖子才发给你。不要再在邮件里回复，点开github再看看是不是跟你沟通的。

**Comment by JungleTrump** [comment]
大佬，PVE里istoreos作旁路由，装了passwall，基本挂一两天就会把临时空间和内存占满，然后就出现各种异常，这是普遍问题还是我哪里搞错了。

**Reply by weiwei52o** [reply]
阿巴阿巴

**Reply by jjm2473** [reply]
你是不是把文件下载到/tmp了

**Comment by chookbro95** [comment]
R2S 经常 3 天就断网，然后就得重启一次才会恢复有人出现过这个种情况吗？

**Reply by xiaotiantian1132** [reply]
你这个，lan口和wan口互换下就好了，我之前也是

**Comment by WarlockFish** [comment]
```
root@iStoreOS:~# sysctl -w net.ipv4.tcp_congestion_control=bbr
sysctl: write error: No such file or directory
root@iStoreOS:~# cat /etc/sysctl.conf 
# Defaults are configured in /etc/sysctl.d/* and can be customized in this file
```

**Reply by jjm2473** [reply]
什么机型？

**Reply by jjm2473** [reply]
固件版本是什么？

**Reply by WarlockFish** [reply]
pve 虚拟机
Linux version 5.10.201 (build@buildserver) (x86_64-openwrt-linux-musl-gcc (OpenWrt GCC 11.2.0 2023070614) 11.2.0, GNU ld (GNU Binutils) 2.37) #0 SMP Fri Mar 8 07:07:46 2024
iStoreOS 22.03.6 2024031514

**Reply by jjm2473** [reply]
@WarlockFish 你从软件包安装bbr了没？你没看更新说明吗？非rk35xx的需要安装软件包。

> 非rk35xx机型可以在软件包里面搜索 `kmod-tcp-bbr` 安装就可以使用BBR了，无需命令

**Reply by Dragonborn2024** [reply]
R66S不能装BBR  R2S能  应该有些装不了

**Reply by jjm2473** [reply]
@Dragonborn2024 rk3568机型已经固件集成了bbr，也就是不需要安装了，都不看更新说明吗？

**Comment by xieyongjian520** [comment]
我断电重启之后就没网了，但是路由器好像有网，lan口的所有设备都没网有大哥知道啥情况吗

**Comment by mr00ff** [comment]
尊敬的开发者 早/中/晚都好
如题 还需要提供啥
虚拟化: pve 8.1
网卡型号: mt7922
固件版本 iStoreOS 22.03.6 2024031514

```shell
dmesg|grep -i wifi

lspci
```

![3660cc3221e029e81122e3c780446e0](https://github.com/istoreos/istoreos/assets/48547769/98e8b73f-7ed1-4460-b2b1-379376c16819)


![33849205814f9722812f73f8fab2877](https://github.com/istoreos/istoreos/assets/48547769/c64fccf0-3737-46a3-aea0-d4eac4536400)


![image](https://github.com/istoreos/istoreos/assets/48547769/b89a82a6-3c1d-45fb-837d-fdc7a7f4aa62)


补充日志信息 :
```shell
root@iStoreOS:~# dmesg | grep mt7921e
[   11.162318] mt7921e 0000:00:10.0: driver own failed
[   11.181357] mt7921e: probe of 0000:00:10.0 failed with error -5

# 内核日志
Sun Apr 14 18:19:46 2024 kern.err kernel: [   13.159618] mt7921e 0000:00:10.0: Message 0004008a (seq 1) timeout
```

**Reply by jjm2473** [reply]
好像intel的wifi也是不能直通到虚拟机内使用，具体原因不清楚。mt7922可能问题也一样。

**Reply by mr00ff** [reply]
感谢大佬回复,同环境同配置情况 有次成功了.我也忘记咋配置的了..后来重装又不行了.我换win11虚拟机能正常使用7922, 搜了一下文献说是尽可能是6或更高版本内核,搜了些源码 看 openwrt/mt76 近期更新过 是把7922包含在内所以猜想是不是 mt76 需要更新一下呢? 再次感谢大佬开源 让我这样的小菜鸡能快乐

**Reply by jjm2473** [reply]
22.03的mt76最近没更新 https://github.com/openwrt/openwrt/blob/openwrt-22.03/package/kernel/mt76/Makefile 。而且既然intel也有问题，可能要等内核升级才能解决了。

**Comment by kongkong1993** [comment]
<img width="781" alt="b171f01052eb1def0a84b445e854c9c" src="https://github.com/istoreos/istoreos/assets/155075721/0fd3cd42-ba80-40ec-a053-60f09e3167b0">
经常会有个口断开然后5秒后又连上是什么原因

**Comment by diave971** [comment]
24.10版本的rk35xx系列cpu还是一样方法开启bbr吗，我看到软件源里面有kmod-tcp-bbr的包，成功安装上了

**Reply by jjm2473** [reply]
24.10版本的rk35xx系列已经支持openwrt官方的内核模块，可以从软件包那边安装。但是版本升级时，这些内核模块会失效需要重新安装，之后会弄个一键恢复失效的内核模块

**Reply by jjm2473** [reply]
24.10只要从软件包安装kmod-tcp-bbr就会自动开启，不需要再配置
